include/nbl/builtin/hlsl/shapes/beziers.hlsl

```diff
@@ -133,610 +198,205 @@ namespace shapes
                       // Taylor expansion at 0 -> f(x)=f(0)+f'(0)*(x) = 1+½x 
                       // So √(1+w) can be approximated by 1+½w
                       // Simplifying later and the fact that w=4p³/q will result in the following.
-                   x = float2(p3/q, -q - p3/q);
+                   x = float2_t(p3/q, -q - p3/q);
                 }
 
-                float2 uv = sign(x)*pow(abs(x), float2(1.0/3.0,1.0/3.0));
-                float t = uv.x + uv.y - kx;
+                float2_t uv = sign(x)*pow(abs(x), float2_t(1.0/3.0,1.0/3.0));
+                float_t t = uv.x + uv.y - kx;
                 t = clamp( t, 0.0, 1.0 );
 
                 // 1 root
-                float2 qos = c + (2.0*a + b*t)*t;
-                res = float2( length(qos),t);
+                float2_t qos = CsubPos + (B + A*t)*t;
+                res = float2_t( length(qos),t);
             }
             else
             {
-                float z = sqrt(-p);
-                float v = acos( q/(p*z*2.0) ) / 3.0;
-                float m = cos(v);
-                float n = sin(v)*1.732050808;
-                float3 t = float3(m + m, -n - m, n - m) * z - kx;
+                float_t z = sqrt(-p);
+                float_t v = acos( q/(p*z*2.0) ) / 3.0;
+                float_t m = cos(v);
+                float_t n = sin(v)*1.732050808;
+                float3_t t = float3_t(m + m, -n - m, n - m) * z - kx;
                 t = clamp( t, 0.0, 1.0 );
 
                 // 3 roots
-                float2 qos = c + (2.0*a + b*t.x)*t.x;
-                float dis = dot(qos,qos);
+                float2_t qos = CsubPos + (B + A*t.x)*t.x;
+                float_t dis = dot(qos,qos);
                 
-                res = float2(dis,t.x);
+                res = float2_t(dis,t.x);
 
-                qos = c + (2.0*a + b*t.y)*t.y;
+                qos = CsubPos + (B + A*t.y)*t.y;
                 dis = dot(qos,qos);
-                if( dis<res.x ) res = float2(dis,t.y );
+                if( dis<res.x ) res = float2_t(dis,t.y );
 
-                qos = c + (2.0*a + b*t.z)*t.z;
+                qos = CsubPos + (B + A*t.z)*t.z;
                 dis = dot(qos,qos);
-                if( dis<res.x ) res = float2(dis,t.z );
+                if( dis<res.x ) res = float2_t(dis,t.z );
 
                 res.x = sqrt( res.x );
             }
             
-            return res.x;
+            return res;
         }
+        
+        template<typename Clipper>
+        float2 ud2(float2_t pos, ArcLengthPrecomputedValues preCompValues, Clipper clipper)
```

Comment:
`ArcLengthPrecomputedValues` should be now named `QuadBezierAnalyticArcLengthCalculator`
https://github.com/Devsh-Graphics-Programming/Nabla-Examples-and-Tests/pull/53/files#r1315761392

remove this from inputs to ud2, the clipper should have a reference to the `QuadBezierAnalyticArcLengthCalculator`

Reply:
done